### ISSUE TITLE
Indefinite hang when calling ToOptimizedResult with a content type of html/text

### ISSUE THREAD

**Issue by deipax** [ServiceStack/Issues#600]
We just converted to ServiceStack5 on my team and I am heading up the conversion work from 3.9.71.  When using a browser to test an endpoint, it hangs indefinitely.  I found the unit test in the source called Can_optimize_html_result_with_ToOptimizedResult, and that works just fine so I know you are testing for it.  I am not exactly sure what is going on with the Task being returned from HTML format ...

To help show the problem, I attached the source for a small service.  Restore the nuget packages and then use a browser to attempt to navigate to: http://localhost:53337/Hello?Name=John

[HelloWorld.zip](https://github.com/ServiceStack/Issues/files/1793366/HelloWorld.zip)

**Comment by mythz** [comment]
The issue was due to our internal re-factor to in v5 to rewrite our internals to avoid sync response writes, but we needed to block here to maintain `ToOptimizedResult()` existing sync signature, unfortunately blocking on writing to the Compression Stream causes a deadlock so I've changed it to block and buffer on a MemoryStream instead which resolves the issue `ToOptimizedResult()` API, but since buffering is unoptimal I've added a new ideal `ToOptimizedResultAsync()` API which doesn't block or buffer. You can use either API but the async API is preferred, e.g:

```csharp
public async Task<object> Get(HelloReq req)
{
    string name = !string.IsNullOrEmpty(req.Name) ? req.Name : "John Smith";
    HelloRes res = new HelloRes()
    {
        Message = $"Hello {0}!"
    };

    return await Request.ToOptimizedResultAsync(res);
}
```

This change is available from v5.0.3 which is now [available on MyGet](http://docs.servicestack.net/myget).

**Comment by deipax** [comment]
Fantastic!  Thank you for the quick turn around mythz.
